auction-server/src/opportunity/api.rs

```diff
@@ -102,148 +183,189 @@ pub struct OpportunityParamsV1 {
+    /// Creation time of the opportunity (in microseconds since the Unix epoch)
+    #[schema(example = 1_700_000_000_000_000i128, value_type = i128)]
+    pub creation_time:  UnixTimestampMicros,
+    #[serde(flatten)]
+    // expands params into component fields in the generated client schemas
+    #[schema(inline)]
+    pub params:         OpportunityCreateEvm,
+}
+
+// ----- Svm types -----
+#[serde_as]
+#[derive(Serialize, Deserialize, ToSchema, Clone, PartialEq, Debug, ToResponse)]
+pub struct TokenAmountSvm {
+    /// Token contract address
+    #[schema(example = "DUcTi3rDyS5QEmZ4BNRBejtArmDCWaPYGfN44vBJXKL5", value_type = String)]
+    #[serde_as(as = "DisplayFromStr")]
+    pub token:  Pubkey,
+    /// Token amount in lamports
+    #[schema(example = 1000)]
+    pub amount: u64,
+}
+
+/// Kamino client specific parameters for the opportunity
+/// It contains the Kamino order to be executed, encoded in base64
+/// SDKs will decode this order and create transaction for bidding on the opportunity
+#[serde_as]
+#[derive(Serialize, Deserialize, ToSchema, Clone, PartialEq, Debug)]
+pub struct OpportunityCreateClientParamsV1KaminoSvm {
+    /// The Kamino order to be executed, encoded in base64
+    #[schema(example = "DUcTi3rDyS5QEmZ4BNRBejtArmDCWaPYGfN44vBJXKL5", value_type = String)]
+    #[serde_as(as = "Base64")]
+    pub order: Vec<u8>,
+}
+
+/// Client specific parameters for the opportunity
+#[derive(Serialize, Deserialize, ToSchema, Clone, PartialEq, Debug)]
+#[serde(tag = "client")]
+pub enum OpportunityCreateClientParamsV1Svm {
+    #[serde(rename = "kamino")]
+    Kamino(OpportunityCreateClientParamsV1KaminoSvm),
+}
+
+/// Opportunity parameters needed for on-chain execution.
+/// Parameters may differ for each client
+#[serde_as]
+#[derive(Serialize, Deserialize, ToSchema, Clone, PartialEq, Debug)]
+pub struct OpportunityCreateV1Svm {
+    /// The permission account to be permitted by the ER contract for the opportunity execution of the protocol
+    #[schema(example = "DUcTi3rDyS5QEmZ4BNRBejtArmDCWaPYGfN44vBJXKL5", value_type = String)]
+    #[serde_as(as = "DisplayFromStr")]
+    pub permission: Pubkey,
+    /// The router account to be used for the opportunity execution of the protocol
+    #[schema(example = "DUcTi3rDyS5QEmZ4BNRBejtArmDCWaPYGfN44vBJXKL5", value_type = String)]
+    #[serde_as(as = "DisplayFromStr")]
+    pub router:     Pubkey,
+    /// The chain id where the opportunity will be executed.
+    #[schema(example = "solana", value_type = String)]
+    pub chain_id:   ChainId,
+    /// The block hash to be used for the opportunity execution
+    #[schema(example = "DUcTi3rDyS5QEmZ4BNRBejtArmDCWaPYGfN44vBJXKL5", value_type = String)]
+    #[serde_as(as = "DisplayFromStr")]
+    pub block_hash: Hash,
+
+    pub sell_tokens: Vec<TokenAmountSvm>,
+    pub buy_tokens:  Vec<TokenAmountSvm>,
+
+    #[serde(flatten)]
+    #[schema(inline)]
+    pub client_params: OpportunityCreateClientParamsV1Svm,
```

Comment:
maybe `program_params`? client feels a bit overused as a term, this field is for program-specific (e.g. `limo`) content